float32 thrust
float32 destheta
float32 desphi
float32 despsi
float32 destheta_p
float32 desphi_p
float32 despsi_p